src/main/java/pl/coderslab/projectwybankuj/web/BankController.java

```diff
@@ -35,11 +45,16 @@ public String addInitForm(Model model) {
     }
 
     @PostMapping("/add")
-    public String addPostForm(@Valid Bank bank, BindingResult bindingResult) {
+    public String addPostForm(@RequestParam("file") MultipartFile file, @Valid Bank bank, BindingResult bindingResult) throws IOException {
         if (bindingResult.hasErrors()) {
             return "addbank";
         }
-        bankRepository.save(bank);
+        String logoName = "logo" + bank.getBankName() + ".jpg";
+        String filePath = request.getSession().getServletContext().getRealPath("/img");
+        File fileToUpload = new File(filePath + "/" + logoName);
+        file.transferTo(fileToUpload);
+        Bank bankToSave = new Bank(bank.getBankName(), logoName);
+        bankRepository.save(bankToSave);
```

Comment:
Za dużo się dzieje w tym kontrolerze oraz użyte jest repozytorium a to powinno byc w serwisie, do którego powinny być testy